cluster/log-dump/log-dump.sh

```diff
@@ -306,7 +306,7 @@ function dump_nodes_with_logexporter() {
   if ! "${KUBECTL}" create -f "${KUBE_ROOT}/cluster/log-dump/logexporter-daemonset.yaml"; then
     echo "Failed to create logexporter daemonset.. falling back to logdump through SSH"
     "${KUBECTL}" delete namespace "${logexporter_namespace}" || true
-    dump_nodes "${NODE_NAMES[@]}"
+    dump_nodes "${NODE_NAMES[@]:+${NODE_NAMES[@]}}}"
```

Comment:
Here also, instead of making this change at 3 places inside the function. I would suggest changing the above condition in L286 to:

```
if [[ -z "${NODE_NAMES:-}" ]]; then
  echo "No nodes found!"
  return
fi
```